dev/io.openliberty.microprofile.reactive.streams.operators30.internal/test/src/io/openliberty/microprofile/reactive/streams/test/SimplePubSubTest.java

```diff
@@ -0,0 +1,229 @@
+/*******************************************************************************
+ * Copyright (c) 2019, 2023 IBM Corporation and others.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License 2.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-2.0/
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *******************************************************************************/
+package io.openliberty.microprofile.reactive.streams.test;
+
+import static org.junit.Assert.assertTrue;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.concurrent.BlockingQueue;
+import java.util.concurrent.CompletionStage;
+import java.util.concurrent.LinkedBlockingQueue;
+
+import org.eclipse.microprofile.reactive.streams.operators.CompletionRunner;
+import org.eclipse.microprofile.reactive.streams.operators.PublisherBuilder;
+import org.eclipse.microprofile.reactive.streams.operators.ReactiveStreams;
+import org.eclipse.microprofile.reactive.streams.operators.SubscriberBuilder;
+import org.junit.Test;
+import org.reactivestreams.Publisher;
+import org.reactivestreams.Subscriber;
+import org.reactivestreams.Subscription;
+
+/**
+ *
+ */
+public class SimplePubSubTest extends AbstractReactiveUnitTest {
```

Comment:
Javadoc explaining the purpose of this test would be helpful. I assume it's a nothing-up-my-sleeves test - ensuring that we can correctly work with a custom `Publisher` or `Subscriber` object and we're not secretly relying on all publishers being a mutiny implementation.